plugins/modules/source_control/gitlab/gitlab_project_variable.py

```diff
@@ -162,106 +241,147 @@ def list_all_project_variables(self):
             vars_page = self.project.variables.list(page=page_nb)
         return variables
 
-    def create_variable(self, key, value, masked, protected, variable_type, environment_scope):
+    def create_variable(self, var_obj):
         if self._module.check_mode:
-            return
+            return True
         var = {
-            "key": key, "value": value,
-            "masked": masked, "protected": protected,
-            "variable_type": variable_type,
+            "key": var_obj.get('key'), "value": var_obj.get('value'),
+            "masked": var_obj.get('masked'), "protected": var_obj.get('protected'),
+            "variable_type": var_obj.get('variable_type'), "environment_scope": var_obj.get('environment_scope')
```

Comment:
IIRC some versions barf if `environment_scope` is passed in. So better set it only when needed, like the code did before.

Reply:
When it is not set, it's `'*'` by GitLab.  
But yes, because of https://github.com/ansible-collections/community.general/pull/4038#discussion_r793349327 it might result in exception on GitLab CE.  
But I'm not sure if a GitLab CE instance quietly accept it, but you just won't see it in the GUI and it has no effect on your pipelines.

I will try against a GitLab CE version later.

Reply:
Some folks might also run very old versions of GitLab (for whatever reason) that were working fine with the code before this PR. We can require a newer version at some point, but not in a minor release.